modules/data-streams/src/main/java/org/elasticsearch/datastreams/rest/RestModifyDataStreamsAction.java

```diff
@@ -40,13 +39,18 @@ public List<Route> routes() {
     protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) throws IOException {
         ModifyDataStreamsAction.Request modifyDsRequest;
         try (XContentParser parser = request.contentParser()) {
-            modifyDsRequest = ModifyDataStreamsAction.Request.PARSER.parse(parser, null);
+            modifyDsRequest = ModifyDataStreamsAction.Request.PARSER.parse(
+                parser,
+                actions -> new ModifyDataStreamsAction.Request(
+                    RestUtils.getMasterNodeTimeout(request),
+                    RestUtils.getAckTimeout(request),
+                    actions
```

Comment:
No need to mutate the request, can just create it with all the right properties up-front.